contracts/MultiMerkleMine.sol

```diff
@@ -0,0 +1,62 @@
+pragma solidity ^0.4.24;
+
+import "./MerkleMine.sol";
+import "zeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "zeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract MultiMerkleMine {
+
+	using SafeMath for uint256;
+
+	bytes[] private _proofs;
+
+	event AlreadyGenerated(address indexed recipient, address indexed caller);
+
+	function extractProofs(bytes _merkleProofs) internal returns(bytes[]) {
+		require(_merkleProofs.length != 0, 'No proofs supplied!');
+		for(uint256 i=0; i<_merkleProofs.length; i++){
+		    uint256 proofSize = uint256(_merkleProofs[i]);
+		    require(proofSize % 32 == 0, 'Invalid proof detected!');
+		    bytes memory _proof = new bytes(proofSize);
+		    uint256 j = 0;
+		    while(j<proofSize){
```

Comment:
@yondonfu @vyomshm do you happen to know if there are lower level assembly opcodes that can do a mem copy more efficiently in terms of gas than looping over each byte in the bytes array one by one?

I don't know that there is, just checking.

Reply:
Inline assembly can be used to perform memory reads/writes of entire 32 byte words using the `mstore` and `mload` instructions which should be a lot more efficient than reading byte by byte.